core/src/signed_envelope.rs

```diff
@@ -44,15 +44,19 @@ impl SignedEnvelope {
         self.key.verify(&buffer, &self.signature)
     }
 
-    /// Extract the payload of this [`SignedEnvelope`].
+    /// Extract the payload and signing key of this [`SignedEnvelope`].
     ///
     /// You must provide the correct domain-separation string and expected payload type in order to get the payload.
     /// This guards against accidental mis-use of the payload where the signature was created for a different purpose or payload type.
+    ///
+    /// It is the caller's responsibility to check that the signing key is what
+    /// is expected. For example, checking that the signing key is from a
+    /// certain peer.
     pub fn payload(
```

Comment:
```suggestion
    pub fn payload_and_signing_key(
```

What do you think of renaming the method as well? If I recall correctly you suggested this in your initial vulnerability report, right?

Reply:
Changed in https://github.com/libp2p/rust-libp2p/pull/2522/commits/ef3afcdeab1b752a52063e3aa3c2be4b356c1474